### DISCUSSION TITLE
CapsuleContainer accessability

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#160] [General]
This code is a Route in `Beamer`:

```dart
Routes.cookies: (context, state, data) => BeamPage(
                key: const ValueKey(Routes.cookies),
                child: const CookiesScreen(),
                title: getTitle('Cookies'),
                type: BeamPageType.fadeTransition,
              ),
```

It would be interesting to have an extension on `BuildContext` to access the `CapsuleContainer`. In this case I need it to access my `languageCapsule` to do some translations here.

**Comment by GregoryConrad** [comment]
Can you wrap the BeamPage in a RearchConsumer? If so, do that.

Not sure what “state” and “data” do, but perhaps you could pass something in there too.

finally, if you can’t do one of the above, you can do a CapsuleContainerProvider.containerOf(context)

**Comment by busslina** [comment]
> Can you wrap the BeamPage in a RearchConsumer?

Don't think so, because `BeamPage` is not a `Widget`.

> Not sure what “state” and “data” do

Seems they hold route info, but also don't think I would be able to prop drill a `CapsuleReader`; and If I could, maybe would be insecure if synchronously is broke. Also prop drilling the wanted value seems to can lead to a much more complex data flow (and seems to be against ReArch philosophy).

> CapsuleContainerProvider.containerOf(context)

So seems I will need this. I understand you didn't create an extension helper to simplify this, because this is not a preferred use, but would you consider to add it with a big warning?

**Reply by GregoryConrad** [reply]
I don't want to introduce an extension because that'd make it too easy to misuse. It should hardly ever be used (and this is one of those rare times)

**Reply by busslina** [reply]
I understand

**Comment by busslina** [comment]
````dart
extension BuildContextExtesion on BuildContext {
  @Deprecated(
    '''Not really insecure, but not recommended to use. Should be only used on a
    situation where some third party code logic prevents to chain it with the
    Widget tree. Routing could be a legitim scenario''',
  )
  CapsuleContainer get insecureCapsuleContainer =>
      CapsuleContainerProvider.containerOf(this);
}
```
